PROVISIONING-METAL.md

```diff
@@ -105,7 +105,30 @@ Please keep in mind that when using static addresses, DNS information must be su
   * `via` (IP address): Gateway IP address.  If no gateway is provided, a scope of `link` is assumed.
   * `route-metric` (integer): Relative route priority.
 
-Example `net.toml` with comments:
+Version `3` adds in support for bonding and vlan tagging. The support is limited to mode `1` (`active-backup`) for [bonding](https://www.kernel.org/doc/Documentation/networking/bonding.txt). Future support may include other bonding options - pull requests are welcome! Version `3` also adds the concept of network devices vs network interfaces. Physical devices like ethernet adapters are considered "interfaces", and network devices encompass other kinds of devices like bonds and vlans. This is a *breaking change* from version 2 and requires the user to specify 
+
+Migrating to version `3` from version `2` requires changes to how devices are defined. For example, if `[eno1]` is the current device name, changing a config from `[eno1]` to `[interface.eno1]` is all that is required. 
+
+Bonding configuration is created via a network device (`device`):
+* `bond` (map): To create a `bond` network device, specify `kind = "bond"`.
```

Comment:
If you're doing an update - please add a blank line before the start of the bullet list. There is another PR that is cleaning up a lot of markdown lint errors. After that merges we could potentially add a markdownlint CI job. If we do that, it will flag this kind of thing where there is no separation before lists and code blocks.